### ISSUE TITLE
Question: Is there a way to assert that an array key exists in a given array?

### ISSUE THREAD

**Issue by boesing** [vimeo/psalm#8785]
Hey there,

I've tried something like this but I don't get a working example:

https://psalm.dev/r/74f4a57aaa

**Comment by orklah** [comment]
I remember looking for something like this for another issue, unfortunately, the internal handling of that is pretty complex in Psalm. We have an HasArrayKey assertion and also an ArrayKeyExists one. If I remember correctly, the HasArrayKey applies to a specific variable and has a $key property for the name of the key that exists, and the ArrayKeyExists directly applies to the `$a['someKey']` identifier.
I'm not really sure why both are needed, I tried to drop ArrayKeyExists but reconciliation broke and I wasn't exactly sure what was going on.

If we want to progress on this, we would probably first need to do exactly that. And then have some sort of syntax that allows to assert dynamic keys (like in your example).

**Comment by VincentLanglet** [comment]
> I remember looking for something like this for another issue, unfortunately, the internal handling of that is pretty complex in Psalm

Do you mean https://github.com/psalm/psalm-plugin-phpunit/issues/114 ?

**Comment by orklah** [comment]
Yep, that's the one (though we were using V4 style assertion in Psalm's internal, so my explanation there is a bit outdated)
